### PR TITLE
SDK: Add Shutdown test for Span Processor

### PR DESCRIPTION
**Pull request by agneum**
Add tests for the issue https://github.com/open-telemetry/opentelemetry-go/issues/164

### CODE REVIEW

File: sdk/trace/span_processor_test.go [hunk 1]
```diff
@@ -109,8 +109,20 @@ func TestUnregisterSpanProcessorWhileSpanIsActive(t *testing.T) {
...
+		t.Fatalf("Error creating new instance of TestSpanProcessor\n")
+	}
+
+	wantCount := sp.shutdownCount + 1
+	sp.Shutdown()
+
+	gotCount := sp.shutdownCount
+	if wantCount != gotCount {
+		t.Errorf("%s: wrong counter: got %d, want %d\n", name, gotCount, wantCount)
+	}
```
Comment (rghetia) [inline sdk/trace/span_processor_test.go]: maybe test Shutdown using Register/Unregister. Multiple Unregister should not trigger multiple shutdown call.

File: sdk/trace/batch_span_processor_test.go [hunk 2]
```diff
@@ -201,4 +201,20 @@ func getSpanContext() core.SpanContext {
...
+		t.Errorf("Unexpected error while creating processor\n")
+	}
+
+	if bsp == nil {
+		t.Fatalf("Error creating new instance of BatchSpanProcessor\n")
+	}
+
+	bsp.Shutdown()
+
+	// Check if it only executes once
```
Comment (rghetia) [inline sdk/trace/batch_span_processor_test.go]: nit: Since there is no actual check here maybe rephrase the comment to "Multiple call to Shutdown() should not panic."

File: sdk/trace/span_processor_test.go (hunk 1 above)
Comment (agneum) [reply sdk/trace/span_processor_test.go]: But actually it triggers. I added another test demonstrating this.
Should I create an issue to fix this behavior?

File: sdk/trace/batch_span_processor_test.go (hunk 2 above)
Comment (agneum) [reply sdk/trace/batch_span_processor_test.go]: Thank you, I fixed the comment.

File: sdk/trace/span_processor_test.go (hunk 1 above)
Comment (rghetia) [reply sdk/trace/span_processor_test.go]: yes, please.

Comment (sjkaris) [reply sdk/trace/span_processor_test.go]: Also, you can link the issue in a comment, so that whoever addresses the issue definitively knows that this test breaking is expected

Comment (agneum) [reply sdk/trace/span_processor_test.go]: I created a new issue: https://github.com/open-telemetry/opentelemetry-go/issues/175
I can fix it, I'm going to add a new sync.Once variable to span_processor.go
